ducktape/cluster/remoteaccount.py

```diff
@@ -142,97 +317,182 @@ def kill_process(self, process_grep_str, clean_shutdown=True, allow_fail=False):
-        if self.user:
-            remotehost = self.user + "@" + self.hostname
-        else:
-            remotehost = self.hostname
+    def copy_between(self, src, dest, dest_node):
+        """Copy src_path to dest_path on dest_node
 
-        if isinstance(src, basestring):
-            src = remotehost + ":" + src
-        else:
-            # assume src is iterable
-            # e.g. "ubuntu@host:path1 ubuntu@host:path2"
-            src = " ".join([remotehost + ":" + path for path in src])
+        :param src_path Path to the file or directory we want to copy
+        :param dest_path The destination path
+        :param dest_node The node to which we want to copy the file/directory
 
-        r = "scp "
-        if self.ssh_args:
-            r += self.ssh_args + " "
-        if recursive:
-            r += "-r "
+        Note that if src is a directory, this will automatically copy recursively.
 
-        r += src + " " + dest
-        return r
+        Example:
 
-    def scp_from(self, src, dest, recursive=False):
-        """Copy something from this node. src may be a string or an iterable of several sources."""
-        return self._ssh_quiet(self.scp_from_command(src, dest, recursive))
+        path/to/file, path/to/renamed_file, node
+            file will be copied to renamed_file on node
 
-    def scp_to_command(self, src, dest, recursive=False):
-        if not isinstance(src, basestring):
-            # Assume src is iterable
-            src = " ".join(src)
+        """
+        # TODO: if dest is an existing file, what is the behavior?
 
-        r = "scp "
-        if self.ssh_args:
-            r += self.ssh_args + " "
-        if recursive:
-            r += "-r "
-        r += src + " "
-        if self.user:
-            r += self.user + "@"
-        r += self.hostname + ":" + dest
-        return r
+        temp_dir = tempfile.mkdtemp()
 
-    def scp_to(self, src, dest, recursive=False):
-        return self._ssh_quiet(self.scp_to_command(src, dest, recursive))
+        try:
+            src_name = src
+            if src_name.endswith(os.path.sep):
+                src_name = src_name[:-len(os.path.sep)]  # trim off path separator from end
+            src_name = os.path.basename(src_name)
+
+            # TODO: deal with very unlikely case that src_name matches temp_dir name?
+            # TODO: I think this actually works
+
+            local_dest = os.path.join(temp_dir, src_name)
+            self.copy_from(src, local_dest)
+
+            dest_node.account.copy_to(local_dest, dest)
+
+        finally:
+            if os.path.isdir(temp_dir):
+                shutil.rmtree(temp_dir)
+
+    def copy_from(self, src, dest):
+        if os.path.isdir(dest):
+            # dest is an existing directory, so assuming src looks like path/to/src_name,
+            # in this case we'll copy as:
+            #   path/to/src_name -> dest/src_name
+            src_name = src
```

Comment:
This code is repeated a bunch of times, could be refactored to make each implementation simpler.